examples/lines_bars_and_markers/bar_label_demo.py

```diff
@@ -0,0 +1,137 @@
+"""
```

Comment:
Why not update the existing example?

Reply:
applicable for `lines_bars_and_markers/barchart` and `horizontal_barchart_distribution`
but sadly, this PR does not provide fine-grained controls needed in `statistics/barchart_demo`
this PR acts on `BarContainer` as a whole instead of fine-tuning color/position for every single bar
and for `misc/demo_ribbon_box` it will not work because they are not bars, they are images